### PR TITLE
Changes for import/export to allow forge and specify world directory

### PR DESCRIPTION
**Pull request by LimpingNinja**
-  Removed journalFileSystem.js - The main reason it wasn't working with Forge was the reimplementation here. Modified all the calls to just directly use FilePicker
- Added ImportWorldPath and ExportWorldPath; labelled them as [Advanced] config options. This allows users to set a custom name beyond the world name but also to select a different folder to Import from. This was mainly done to cross-pollinate worlds.
-  Added parse function from VTTA Tokenizer to parse out the string-based token. **Might be good to change to a directory picker here, but I'll leave it alone for now**. If someone wants to use Forge they can prepend [forgevtt] on the string.
- Modified importFile to respect URL pathing for file names and appropriately modified the call before fetch.

This is probably not as clean as they should come. I would request local testing against a local installation just to be sure it all works. My local installation still uses forge storage. 

Additionally, this has been tested on Forge, local with Forge module, and cross-world. A big **caveat** with cross-world is that journal links will break, this is anticipated. A future enhancement here would be to drop out the mapping of path{name,old-id} as a json file at the root of worldsource path when creating the export and then have a /js resync [file] that uses the specified file to find references to the old id and replace them with the new id.

### CODE REVIEW

**Comment by sytone** [comment]
Change looks good. I was considering adding a JSON file to help with metadata mapping or adding a YAML header to the markdown files which would allow for additional properties to be persisted to the files from the journal where the journal content was not a JSON file already.

**Comment by sytone** [comment]
Tried a export with the advanced export option and hit a error

journal-sync | INFO |  ENOENT: no such file or directory, mkdir 'f:\foundry\data\Data\journal-sync\defaultexportpath\Blackhollow'

Which exposed a bug where the root folder was not being created. :)

In export folder there also need to be a create Directory for the parent to make sure it exists. We can just duplicate the call which is a bit brute force but would fix it for the moment.

File: module/journal-sync.js [hunk 1]
```diff
@@ -320,7 +372,7 @@ async function exportFolder(folder, parentPath) {
     let folderPath = (parentPath + '/' + folder.data.name).replace("//", "/").trim();
 
     // Create folder directory on server. 
-    FS.createDirectory("data", folderPath)
+    FilePicker.createDirectory(markdownPathOptions.activeSource, folderPath)
```
Comment (sytone) [inline module/journal-sync.js:376]: folderPath is a combination of parent and folder. Looks like there is no check for parent folder to see if it exists. Create directory could just be duplicated for a parent check.

Comment (LimpingNinja) [reply module/journal-sync.js:376]: Fair enough, I realize I missed that on my internal version yesterday night as well as I was making a poor assumption on world's existing from a workflow view. MVP based on tightly scoped requirements? ;-) I'll fix it.

Comment (LimpingNinja) [reply module/journal-sync.js:376]: Side note, in testing I don't see that the catch on FilePicker.createDirectory is every really called. It looks like FilePicker.createDirectory is fully non-destructive and non-returning in the case of existence (checked for by EEXIST)... just a note.

Comment (sytone) [reply module/journal-sync.js:376]: Yea, I think I was just trying to catch the exception in a nicer way compared to another error I was seeing at the time.

**Comment by LimpingNinja** [comment]
Change resolved, let me know if you see anything else. I'm looking at that function as per conversation but didn't feel a need to refactor it for this change.

**Comment by sytone** [comment]
Will look at it today. Thanks!

**Pull request merged**